Sources/NIO/Thread.swift

```diff
@@ -176,21 +249,66 @@ public final class ThreadSpecificVariable<Value: AnyObject> {
     /* the actual type in there is `Box<(ThreadSpecificVariable<T>, T)>` but we can't use that as C functions can't capture (even types) */
     private typealias BoxedType = Box<(AnyObject, AnyObject)>
 
-    private let key: pthread_key_t
+    private class Key {
```

Comment:
Are we making use of `Key` being reference counted? Aren't we just wrapping a handle which is destroyed explicitly and not though a `deinit`? I think we should make this a `struct Key`

Reply:
Ugh, you're right.  I had overlooked the manual key destruction when I first wrote that, and only realized that later (since so much code was blindly removed) and forgot to update this.

Reply:
Oh, wait, I think Im mixing up another site.  Here, the original code was adding an explicit `deinit` that deleted the key.  I can certainly make the wrapper a struct and add a static method to just dispatch the delete.

Reply:
sorry, I think this is fine. If we do use a `deinit`, then `class` is great. Thought the code wasn't using `deinit` nor any of the reference properties and it was still a `class` but I think I misread.

Reply:
Yes, the `deinit` destroys the slot.  Its a tradeoff ... now the slot itself is managed, so now `ThreadSpecificVariable` could be a `struct`, but its `public` so I didn't want to touch it.  IMO this is nicer because it reflects the reality - the value itself is not what was refcounted, it was the slot.